### ISSUE TITLE
📣 Upgrade your integration to continue accepting Braintree payments

### ISSUE THREAD

**Issue by saralvasquez** [braintree/braintree_php#349]
# Upgrade your integration to continue accepting Braintree payments

Braintree updated the root SSL certificate provider for api.braintreegateway.com (sandbox and prod) on April 12, 2024. The SSL certificates for PHP SDK versions older than 6.21.0 are set to expire by June 31, 2025.

If you do not update your SDK to the latest version with the updated certificates by June 31, 2025, 100% of your impacted traffic  will fail.

To reduce the impact, your developer will need to update your SDK to version 6.21.0 at a minimum, though using the latest version of the SDK is recommended. For details on how to setup the SDK, see our setup guide found [here](https://developer.paypal.com/braintree/docs/start/hello-server/java).

### What other SDKs will be impacted?

The following SDKs need to be updated to the latest version by June 31, 2025. Check the minimum server versions in the [Braintree dev docs](https://developer.paypal.com/braintree/docs/reference/general/best-practices/ruby#server-sdk-versions).

- Android v4
- iOS v5, v6
- Java v3
- Python v4
- Ruby v4

### When do I need to make this change?  
By June 31, 2025.

### Is there any flexibility in the deadline?
No. Please work with your TAM to ensure you will meet the deadline in time.

### What if I’m using an older version? 
Only version 6.21.0 or newer will contain the updated SSL certificates. Any older versions must migrate to this version by June 31, 2025 to avoid impact of traffic to Braintree SDKs.

### What changes does my developer need to make to my integration?  
Your developers needs to update the SDK (client and server) to the latest version to eliminate customer service disruptions and for good security practice.

Braintree's official recommendation is that Merchants use official SDKs, not modified ones. We cannot support modified SDKs. Reference our docs for our official SDK - https://developer.paypal.com/braintree/docs/

It is suggested to make this update as soon as possible to reduce any impacts to the customer experience.  

### I've updated my SDK but I'm still getting emails to update
Braintree regularly pulls the latest list of impacted Merchants who need to update their SDK version before sending each email. If you have communicated with your TAM that you will updated but have not, then you will continue to receive email reminders until you have completed the update.

### What happens if I don’t update or miss the deadline?
If you are still processing on a legacy SDK when our SSL Root Certificate update goes live, your API calls will no longer be able to reach Braintree's servers. Your customers will start to see errors when trying to complete checkout and their app logs will show SSL-related errors.

As a result, the specific error message you encounter won't be from Braintree, but rather from the language or framework you are using. The error message will likely be unique to that language or framework, but you can expect it to mention SSL – something along the lines of an invalid certificate, a refused connection, a failed handshake, etc. If you begin to encounter an error along those lines, double check that your integration has been updated to at least the minimum version, and then feel free to reach back out to us for further support.

### What region is impacted by this change?
All (global)

**Comment by urifoox** [comment]
Hello - what specifically changed? Is it the Gene/Braintree/braintree/braintree_php/lib/ssl/api_braintreegateway_com.ca.crt file or is it something else? Is there a diff we can compare for PHP of what we have and what we need to update?

**Comment by saralvasquez** [comment]
Yeah exactly! The only change is that new valid certificates were added to the lib/ssl/api_braintreegateway_com.ca.crt file. If you are using version 6.21.0 or newer then you're all set! If you wanted to investigate the changes more closely, looking through the history of that file should show you everything that changed.

**Comment by urifoox** [comment]
Can we just change that file and nothing else? We’re not looking to upgrade
the entire SDK. Everything is tested and working with the current version
we’re using and updating the sdk may introduce a ton of issues

**Comment by saralvasquez** [comment]
After the cut over in June we will no longer support older versions of the SDK so you're going to need to upgrade to that new minimum version no matter what. I understand concern for breaking your integration when it's working as you'd like, but we do adhere to semantic versioning and all the currently supported versions of the SDK are within the same major version. So you can assume nothing should break by updating to any of the newer versions

**Comment by urifoox** [comment]
I appreciate that there won't be support for the older versions of the SDK but other than this one file change will anything else actually break? We can schedule a time on our roadmap to upgrade the SDK but is it an absolute requirement to hit this deadline to upgrade? I don't want to conflate two things together if they are separate. 

There's an enormous time/risk difference between updating 1 file and updating the entire SDK. We're happy to update that file (how can we test the change works?) and push out a fix but if that won't work, please advise. 

For context, we are a B2B SaaS eCommerce Platform representing hundreds of customers. We received no communication about this until today from one of our customers who received an email which a Google thread popped this issue up. We're happy to continue working as partners but please understand the difference in scope you're asking of our technical team mid roadmap.

**Comment by EugenOrosan** [comment]
Just received today an email that the SDK for PHP will need to be updated to at least 6.21.0 version. We are using it in the Magento implementation and for the latest Magento version release, the Braintree SDK version used is 6.13.0. 

Do you know if there are some updates related to the module implementation for Magento?

In the email that we received, the date mentioned is June 30.

**Comment by saralvasquez** [comment]
> I appreciate that there won't be support for the older versions of the SDK but other than this one file change will anything else actually break? We can schedule a time on our roadmap to upgrade the SDK but is it an absolute requirement to hit this deadline to upgrade? I don't want to conflate two things together if they are separate.
> 
> There's an enormous time/risk difference between updating 1 file and updating the entire SDK. We're happy to update that file (how can we test the change works?) and push out a fix but if that won't work, please advise.
> 
> For context, we are a B2B SaaS eCommerce Platform representing hundreds of customers. We received no communication about this until today from one of our customers who received an email which a Google thread popped this issue up. We're happy to continue working as partners but please understand the difference in scope you're asking of our technical team mid roadmap.
> 
> Also - not to nitpick but June 31, 2025 (the deadline listed above) is not a date. Do you mean June 30 or July 1?

The two are not separate, the way to maintain access to Braintree during this transition is to upgrade to the new minimum version (6.21.0). Once the SDK has been upgraded to the new version the only testing required is sending requests as normal. If you can send requests and get responses then everything is good to go.

Thanks for catching that! The deadline is June 30th 2025

**Comment by saralvasquez** [comment]
> Just received today an email that the SDK for PHP will need to be updated to at least 6.21.0 version. We are using it in the Magento implementation and for the latest Magento version release, the Braintree SDK version used is 6.13.0.
> 
> Do you know if there are some updates related to the module implementation for Magento?
> 
> In the email that we received, the date mentioned is June 30.

We only have context on this SDK directly, so we don't have any information on other software products that utilize the SDK. I'd recommend reaching out to Magento directly for support

**Comment by sergei-ft** [comment]
June 31, 2025 doesn't exist. Please, fix the date in the issue and in the README in repo.
